client.go

```diff
@@ -0,0 +1,138 @@
+	logger.WithField("error", err).Error("[METRICS] Error caught in metrics")
+}
+
+/*
+MetricsSender is a struct responsible for keeping all the statsd clients
+*/
+type MetricsSender struct {
+	Clients []*statsd.Client
+}
+
+/*
+Array of metricDestination (retrieve each metric destination from METRIC_DESTINATIONS)
+*/
+type metricsDestinations []*metricsDestination
+
+/*
+Struct target from METRICS_DESTINATION env var
+*/
+type metricsDestination struct {
+	Host   string `json:"METRICS_HOST"`
+	Port   string `json:"METRICS_PORT"`
+	Prefix string `json:"METRICS_PREFIX"`
+}
+
+/*
+The config to pass to a client builder
+*/
+type clientConfig struct {
+	Host, Prefix string
+}
+
+/*
+Return a client for a metric destination struct
+*/
+func clientConfigFromDestination(destination *metricsDestination) *clientConfig {
+	return &clientConfig{
+		Host:   destination.Host + ":" + destination.Port,
+		Prefix: destination.Prefix,
+	}
+}
+
+/*
+Struct responsible for building clients
+*/
+type clientBuilder struct {
+	errorLogger func(error)
+}
+
+/*
+Return a statsd client from our custom clientConfig
+statsd.new will resolve the given host and panics if it fails
+The statsd lib we use sends a message upon connection
+if there is nothing listening it may crash
+TODO : check that this code is removed https://github.com/alexcesaro/statsd/blob/master/conn.go#L50
+*/
+func (builder *clientBuilder) buildClient(config *clientConfig) *statsd.Client {
+	client, err := statsd.New(
+		statsd.Address(config.Host),
+		statsd.Prefix(config.Prefix),
+		statsd.ErrorHandler(builder.errorLogger),
+	)
+
+	if err != nil {
+		logger.WithFields(logrus.Fields{
+			"config": config,
+			"error":  err,
+		}).Error("Error creating the statsd client")
+		panic("Error creating statsd client")
+	}
+
+	return client
+}
+
+/*
+Initialize the client from the standard configuration of metrics (METRICS_HOST, METRICS_PORT, METRICS_PREFIX)
+*/
+func (metricsSender *MetricsSender) initStandardClient(clientBuilder *clientBuilder) {
+	host := os.Getenv("METRICS_HOST")
+
+	if host == "" {
+		logger.Info("[METRICS] METRICS_HOST empty, not initializing a client for the standard configuration")
+	} else {
+		port := os.Getenv("METRICS_PORT")
+		prefix := os.Getenv("METRICS_PREFIX")
+
+		if port == "" || prefix == "" {
+			panic("[METRICS] Basic configuration can not have any empty port or prefix")
+		}
+
+		basicConfig := &clientConfig{
+			Host:   host + ":" + port,
+			Prefix: prefix,
+		}
+		metricsSender.Clients = append(metricsSender.Clients, clientBuilder.buildClient(basicConfig))
+	}
+}
+
+/*
+Initialize one or several clients from the METRICS_DESTINATIONS env var
+*/
+func (metricsSender *MetricsSender) initDestinationClients(clientBuilder *clientBuilder) {
+	metricsDestinationVar := os.Getenv("METRICS_DESTINATIONS")
+
+	if metricsDestinationVar == "" {
+		logger.Info("[METRICS] METRICS_DESTINATIONS empty, not initializing a client for the advanced configuration")
+	} else {
```

Comment:
agreed, this felt like a nodejs way of doing thing but keeping functions less indented is ultimately clearer